recipes/libcugraph/meta.yaml

```diff
@@ -0,0 +1,53 @@
+{% set name = "libcugraph" %}
+{% set version = "0.19.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/rapidsai/cugraph/archive/refs/tags/v{{ version }}a.tar.gz
+    sha256: eea8eb788ee2e9f369b6d4911c8cb7c4819453e17aa855eac14af86ee27f627e
+    folder: cugraph
+  # libcugraph depends on the RMM header-only library of the same version.
+  - url: https://github.com/rapidsai/rmm/archive/refs/tags/v{{ version }}a.tar.gz
+    sha256: aa2b6542e51e9d3bec1c4053ab2a06af8fb8e9a94be3832418109904ebebb276
+    folder: rmm
+
+build:
+  number: 0
+  skip: true  # [(not linux) or (cuda_compiler_version == "None")]
+  run_exports:
+    - {{ pin_subpackage(name, max_pin="x.x") }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}
+    - cmake >=3.12.4
+    - make
+  host:
+    - boost-cpp >=1.66
+    - nccl >=2.8.4
+    - ucx-proc =*=gpu
+    - faiss-proc =*=cuda
+    - libfaiss =1.7.0=*_cuda
```

Comment:
@h-vetinari do you know the right way to list these as requirements? We seem to be running into [solver errors on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=303513&view=logs&j=232b7f5e-8d64-5438-fada-8478f379d226&t=e8516d7e-1153-5090-5807-53ae833fddd8 )

Reply:
That's interesting, because that's exactly how the dependency is also specified for `faiss -> libfaiss` itself: https://github.com/conda-forge/faiss-split-feedstock/blob/master/recipe/meta.yaml#L202

Reply:
Can you try without the `faiss-proc`? That's redundant anyway to using the build-string for `libfaiss`

Reply:
As another data point, the following resolves correctly for me on windows (w/ cuda 11.2):
```
conda create -n test python=3.8 "libfaiss =1.7.0=*_cuda" "faiss-proc=*=cuda"
```

Reply:
I see this still failed. As a workaround, you can try using the `faiss-gpu` compatibility package.

There's also a small chance that I broke something with https://github.com/conda-forge/faiss-split-feedstock/pull/42, but it did seem quite straight-forward from https://github.com/conda-forge/pypy3.6-feedstock/blob/master/recipe/meta.yaml#L66-L90 🤷

Reply:
Thanks for looking at this @h-vetinari .  I noticed that it does find `libfaiss` as seen in the output just above the failure:
```
The following NEW packages will be INSTALLED:
    ...
    libfaiss:       1.7.0-cuda102hd37495c_7_cuda conda-forge
```
but for some reason something else is having a problem elsewhere.  I'll try `faiss-gpu`, but I think I may need to do additional testing on my end to ensure libcugraph still functions correctly with the different dependency.

Reply:
> but I think I may need to do additional testing on my end to ensure libcugraph still functions correctly with the different dependency.

`libfaiss=*=_cuda` and `faiss-gpu` are exactly equivalent.

Reply:
> `libfaiss=*=_cuda` and `faiss-gpu` are exactly equivalent (up to the abstraction of having the minimally necessary conda wrapper)

Apologies, I shouldn't try to multi-task, apparently 😅 

`faiss-gpu` is equivalent to `faiss=*=_cuda` (without the "lib"), which is a _superset_ of `libfaiss=*=_cuda`. I don't understand why `libfaiss` alone cannot be resolved, unless it's somehow a weird consequence of conda/conda-build#4090 (due to `libfaiss` & `libfaiss-avx2` being built by a jinja-loop)

Reply:
Am just guessing, but is it possible that this `run_export` isn't resolved correctly? Wondering this based on the fact that the package is found by the solver initially, but then has install issues later

https://github.com/conda-forge/faiss-split-feedstock/blob/a2fa190a5e11fc0c26647d585e1958a8846c63c6/recipe/meta.yaml#L146

Reply:
Yeah, that sounds like a very likely culprit - seems the "libext" does not get resolved correctly

Reply:
I commented on the conda-build issue: https://github.com/conda/conda-build/issues/4090. Is it feasible for you to use the full `faiss` (rather than `libfaiss`) for the moment, until that issue gets fixed (or alternatively: https://github.com/conda-forge/conda-forge.github.io/issues/1261)?

I'm really reluctant to duplicate 50 lines of code that are almost exactly but not quite the same and then keep them in sync, but could be grudgingly convinced otherwise if necessary.

Reply:
As this is a C++ library, I think we want to skip Python if that's an option

It might be simpler than that. For example what if `'libfaiss{{ libext }}'` were assigned to a variable that is then used in that expression? Or something like this `- {{ pin_compatible('libfaiss%s" % libext) }}`?

Reply:
The builds form https://github.com/conda-forge/faiss-split-feedstock/pull/43 are now available. :)